python/pyspark/sql/connect/plan.py

```diff
@@ -73,9 +73,21 @@ class LogicalPlan:
 
     INDENT = 2
 
-    def __init__(self, child: Optional["LogicalPlan"]) -> None:
+    def __init__(
+        self, child: Optional["LogicalPlan"], references: Optional[Sequence["LogicalPlan"]] = None
+    ) -> None:
         self._child = child
-        self._plan_id = LogicalPlan._fresh_plan_id()
+        self._original_plan_id = LogicalPlan._fresh_plan_id()
```

Comment:
I don't understand why we need a `_original_plan_id`?

Reply:
If the plan is wrapped with `WithRelations`, the `_plan_id` should be the id for `WithRelations`. It just keeps the root plan's id.

Reply:
ah, got it!